File: CarHomeWidget/src/com/anod/car/home/skin/BaseProperties.java
package com.anod.car.home.skin;

import com.anod.car.home.R;

public abstract class BaseProperties implements SkinProperties {

	@Override
	public int getInCarButtonExitRes() {
		return R.drawable.ic_incar_exit;
	}

	@Override
	public int getInCarButtonEnterRes() {
		return R.drawable.ic_incar_enter;
	}
	
	@Override
	public int getSetShortcutRes() {
		return R.drawable.ic_add_shortcut;
	}

	@Override
	public IconProcessor getIconProcessor() {
		return null;
	}

	
	
}

File: CarHomeWidget/src/main/java/com/anod/car/home/skin/BaseProperties.kt
package com.anod.car.home.skin

import com.anod.car.home.R
import com.anod.car.home.skin.icon.BackgroundProcessor
import com.anod.car.home.skin.icon.IconProcessor

abstract class BaseProperties : SkinProperties {

    override val inCarButtonExitRes: Int
        get() = R.drawable.ic_incar_exit

    override val inCarButtonEnterRes: Int
        get() = R.drawable.ic_incar_enter

    override val setShortcutRes: Int
        get() = R.drawable.ic_add_shortcut

    override val iconProcessor: IconProcessor?
        get() = null

    override val setShortcutText: Int
        get() = R.string.set_shortcut

    override val iconPaddingRes: Int
        get() = R.dimen.icon_padding_bottom

    override val backgroundProcessor: BackgroundProcessor?
        get() = null

    override fun hasWidgetButton1(): Boolean {
        return true
    }

}